infrastructure/logging/log_summarizer.cc

```diff
@@ -0,0 +1,67 @@
+//%bin(log_summarizer)
+
+//%deps(yaml-cpp)
+#include <iostream>
+#include <iomanip>
+#include "infrastructure/logging/log_reader.hh"
+#include "infrastructure/config/config.hh"
+#include "infrastructure/time/timestamp.hh"
+#include "embedded/imu_driver/imu_message.hh"
+
+
+struct ChannelStats{
+  jet::Timestamp first_message_received;
+  jet::Timestamp last_message_received;
+  int num_messages_received;
+  double frequency;
+};
+
+std::optional<std::string> parse_commandline_arguments(int argc, char* argv[]) {
+  if (argc > 1) {
+    return argv[1];
+  }
+  return {};
+}
+
+int main(int argc, char* argv[]) {
+  const std::optional<std::string> log_path = parse_commandline_arguments(argc, argv);
+  Config log_config;
+  if (!log_path.has_value()) {
+    std::cout << "Please enter a log path" << std::endl;
+    return 0;
+  }
+
+  jet::LogReader log_reader(*log_path, {});
+  const std::vector<std::string> channels = log_reader.get_available_channels();
+  std::vector<ChannelStats> channel_stats(channels.size());
+
+  // Go through all the messages
+  for (std::size_t i = 0; i < channels.size(); ++i) {
+    jet::Message message = {};
+    if (log_reader.read_next_message(channels[i], message)) {
+      channel_stats[i].first_message_received = message.header.timestamp_ns;
+      channel_stats[i].num_messages_received++;
+    }
+    while(log_reader.read_next_message(channels[i], message)) {
+      channel_stats[i].last_message_received = message.header.timestamp_ns;
+      channel_stats[i].num_messages_received++;
+    }
+    channel_stats[i].frequency = static_cast<double>(channel_stats[i].num_messages_received)/(std::chrono::duration_cast<std::chrono::seconds>(std::chrono::nanoseconds(channel_stats[i].last_message_received - channel_stats[i].first_message_received))).count();
```

Comment:
`experiments/estimation/time_point.hh` has these operations

The ones I think you want are `to_time_point(const Timestamp&)`, and `estimation::to_seconds(const Duration&)`

In any case, you should factor your std::chrono operations into functions like `get_seconds()`